webview-ui/src/components/chat/IndexingStatusBadge.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useState, useEffect, useMemo } from "react"
+import { cn } from "@src/lib/utils"
+import { vscode } from "@src/utils/vscode"
+import { useAppTranslation } from "@/i18n/TranslationContext"
+import { useTooltip } from "@/hooks/useTooltip"
+import type { IndexingStatus, IndexingStatusUpdateMessage } from "@roo/ExtensionMessage"
+
+interface IndexingStatusDotProps {
+	onNavigateToSettings?: () => void
+	className?: string
+}
+
+export const IndexingStatusDot: React.FC<IndexingStatusDotProps> = ({ onNavigateToSettings, className }) => {
+	const { t } = useAppTranslation()
+	const { showTooltip, handleMouseEnter, handleMouseLeave, cleanup } = useTooltip({ delay: 300 })
+	const [isHovered, setIsHovered] = useState(false)
+
+	const [indexingStatus, setIndexingStatus] = useState<IndexingStatus>({
+		systemStatus: "Standby",
+		processedItems: 0,
+		totalItems: 0,
+		currentItemUnit: "items",
+	})
+
+	useEffect(() => {
+		// Request initial indexing status
+		vscode.postMessage({ type: "requestIndexingStatus" })
+
+		// Set up message listener for status updates
+		const handleMessage = (event: MessageEvent<IndexingStatusUpdateMessage>) => {
+			if (event.data.type === "indexingStatusUpdate") {
+				const status = event.data.values
+				setIndexingStatus(status)
+			}
+		}
+
+		window.addEventListener("message", handleMessage)
+
+		return () => {
+			window.removeEventListener("message", handleMessage)
+			cleanup()
+		}
+	}, [cleanup])
+
+	// Calculate progress percentage with memoization
+	const progressPercentage = useMemo(
+		() =>
+			indexingStatus.totalItems > 0
+				? Math.round((indexingStatus.processedItems / indexingStatus.totalItems) * 100)
+				: 0,
+		[indexingStatus.processedItems, indexingStatus.totalItems],
+	)
+
+	// Get tooltip text with internationalization
+	const getTooltipText = () => {
+		switch (indexingStatus.systemStatus) {
+			case "Standby":
+				return t("chat:indexingStatus.ready")
+			case "Indexing":
+				return t("chat:indexingStatus.indexing", { percentage: progressPercentage })
+			case "Indexed":
+				return t("chat:indexingStatus.indexed")
+			case "Error":
+				return t("chat:indexingStatus.error")
+			default:
+				return t("chat:indexingStatus.status")
+		}
+	}
+
+	// Navigate to settings when clicked
+	const handleClick = () => {
+		if (onNavigateToSettings) {
+			onNavigateToSettings()
+		}
```

Comment:
I think you can just do this instead of all of the `onNavigateToSettings` plumbing:

```
window.postMessage(
	{
		type: "action",
		action: "settingsButtonClicked",
		values: { section: "experimental" },
	},
	"*",
)
```